tools/build-tool/castle-engine.dpr

```diff
@@ -414,6 +427,18 @@ begin
           Project.DoRun(Target, OS, CPU, RestOfParameters);
         finally FreeAndNil(RestOfParameters) end;
       end else
+      if Command = 'compile-run' then
+      begin
+        Project.DoCompile(OverrideCompiler, Target, OS, CPU, Mode, CompilerExtraOptions);
+        RestOfParameters := TCastleStringList.Create;
```

Comment:
This is copy-paste of logic done for `run`, better extract it to a function `CreateRunParameters` to be reused both when 

`if Command = 'run' then` 

and when

`if Command = 'compile-run' the`

Like

```
function CreateRunParameters: TCastleStringList;
begin
  Result := TCastleStringList.Create;
  Result.Text := Parameters.Text;
  Result.Delete(0); // remove our own name
  Assert(SameText(Result[0], 'run') or SameText(Result[0], 'compile-run'));
  Result.Delete(0); // remove "run" or "compile-run"
end;
```